src/template.rs

```diff
@@ -55,15 +56,15 @@ impl Signature {
     }
 }
 
-impl FunctionTemplate {
+impl<T> FunctionTemplate<T> {
     /// Creates a function template.
     pub fn new(isolate: &isolate::Isolate,
                context: &context::Context,
-               callback: Box<Fn(value::FunctionCallbackInfo) -> value::Value + 'static>)
-               -> FunctionTemplate {
+               callback: Box<Fn(value::FunctionCallbackInfo<T>) -> value::Value + 'static>)
+               -> FunctionTemplate<T> {
```

Comment:
The inconsistency here is because since the instance is not passed to ObjectTemplate::new it doesn't have any way to infer that type. Adding a type parameter broke existing code that didnt pass one via something like ObjectTemplate::new::<T>(...) whereas arguments passed to FunctionTemplate::new had the necessary type information for it to be inferred. ObjectTemplate::with_internal was added to avoid breaking existing code and to avoid the somewhat clunky default case of ObjectTemplate::new::<()>(...). If there was a way to work around this and specify a default I agree it would be a lot cleaner and more consistent but I couldn't think of one.